src/FakeItEasy.Analyzer.Shared/NonVirtualSetupAnalyzer.cs

```diff
@@ -128,9 +160,55 @@ private static bool IsMethod(SymbolInfo symbolInfo)
             return symbolInfo.Symbol?.Kind == SymbolKind.Method;
         }
 
+        private static bool IsIndexer(SymbolInfo symbolInfo)
+        {
+            if (symbolInfo.Symbol?.Kind == SymbolKind.Property)
+            {
+                var propertySymbol = (IPropertySymbol)symbolInfo.Symbol;
+                return propertySymbol.IsIndexer;
+            }
+
+            return false;
+        }
+
         private static bool IsInterfaceMember(SymbolInfo symbolInfo)
         {
             return symbolInfo.Symbol?.ContainingType?.TypeKind == TypeKind.Interface;
         }
+
+        private static InvocationExpressionSyntax FindInvocationInHierarchy(SyntaxNode node, IEnumerable<Ancestor> ancestors)

```

Comment:
> ancestors sounds to me like these are the actual ancestors, not the ones we require (or expect). Perhaps a name change here and for NodeAncestors?

What do you have in mind? Something like `expectedAncestors` or `requiredAncestors`?

> Now that I notice it, we could opt not to pass in ancestors and just use NodeAncestors directly.

Good point.